src/com/palmergames/bukkit/towny/Towny.java

```diff
@@ -483,183 +477,100 @@ public void onDisable() {
-
-		//Add our chat handler to TheNewChat via the API.
-		if(Bukkit.getPluginManager().isPluginEnabled("TheNewChat")) {
-			TNCRegister.initialize();
-		}
-
-		//Legacy check to see if questioner.jar is still present.
-		test = getServer().getPluginManager().getPlugin("Questioner");
-		if (test != null) {
-			String questioner = "  Warning: Questioner.jar present on server, Towny no longer requires Questioner for invites/confirmations. You may safely remove Questioner.jar from your plugins folder.";
-			plugin.getLogger().warning(StringMgmt.wrap(questioner, 55, System.lineSeparator() + "                           "));
-		}
+		if (Bukkit.getPluginManager().isPluginEnabled("Citizens2"))
+			setCitizens2(true);
 
-		//Add warning about PowerRanks.
-		test = getServer().getPluginManager().getPlugin("PowerRanks");
-		if (test != null) {
-			String powerranks = "  Warning: PowerRanks is incompatible with Towny. PowerRanks will override Towny's ability to give permissions via the townyperms.yml file."
-							+ " You can expect issues with Towny permissions (and other permission providers,) while PowerRanks is installed.";
-			plugin.getLogger().warning(StringMgmt.wrap(powerranks, 55, System.lineSeparator() + "                           "));
+		if (!discovered.isEmpty()) {
+			plugin.getLogger().info(Translation.of("msg_compat_discovered", discovered.size(), String.join(", ", discovered)));
 		}
-
 	}
-	
+
 	private String returnPermissionsProviders() {
-		// TownyPerms is always present.
-		String output = "  Permissions: TownyPerms, ";
-		
-		// Test for GroupManager being present.
-		Plugin test = getServer().getPluginManager().getPlugin("GroupManager");
-		if (test != null) {
-			TownyUniverse.getInstance().setPermissionSource(new GroupManagerSource(this, test));
-			output += String.format("%s v%s", "GroupManager", test.getDescription().getVersion());
-		// Else test for vault being present.
+		StringBuilder builder = new StringBuilder();
+		builder.append("TownyPerms, ");
+
+		// Test the presence of GroupManager.
+		final @Nullable Plugin gm = getServer().getPluginManager().getPlugin("GroupManager");
+		if (gm != null) {
+			TownyUniverse.getInstance().setPermissionSource(new GroupManagerSource(this, gm));
+			builder.append("GroupManager");
 		} else {
-			// Try Vault
-			test = getServer().getPluginManager().getPlugin("Vault");
-			if (test != null) {
-				RegisteredServiceProvider<Chat> chatProvider = findChatImplementation();
-				RegisteredServiceProvider<Permission> permissionProvider = getServer().getServicesManager().getRegistration(Permission.class);
-
-				if (chatProvider == null) {
-					// No Chat implementation
-					test = null;
-					// Fall back to BukkitPermissions below
+			// If GroupManager is not present, test the presence of Vault.
+			final @Nullable Plugin vault = getServer().getPluginManager().getPlugin("Vault");
+			if (vault != null) {
+				RegisteredServiceProvider<Chat> chat = findChatImplementation();
```

Comment:
I'm not certain we actually need to report chat functionality, we're only using the prefix/suffix values that come from permission plugins.

Reply:
I tried to make this the same as the current implementation. by "report ... functionality" do you mean informing via an separate message?